hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/LazyCreateThroughPrecompileSuite.java

```diff
@@ -655,4 +660,62 @@ final HapiSpec htsTransferFromForNFTLazyCreate() {
+        return defaultHapiSpec("revertedAutoCreationRollsBackEvenIfTopLevelSucceeds")
+                .given(
+                        snapshotMode(
+                                FUZZY_MATCH_AGAINST_HAPI_TEST_STREAMS,
+                                NONDETERMINISTIC_FUNCTION_PARAMETERS,
+                                ACCEPTED_MONO_GAS_CALCULATION_DIFFERENCE),
+                        newKeyNamed(ECDSA_KEY).shape(SECP_256K1_SHAPE),
+                        newKeyNamed(MULTI_KEY),
+                        cryptoCreate(OWNER).balance(100 * ONE_HUNDRED_HBARS).maxAutomaticTokenAssociations(5),
+                        tokenCreate(NFT_TOKEN)
+                                .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                                .treasury(OWNER)
+                                .initialSupply(0L)
+                                .supplyKey(MULTI_KEY),
+                        uploadInitCode(AUTO_CREATION_MODES),
+                        contractCreate(AUTO_CREATION_MODES),
+                        mintToken(NFT_TOKEN, List.of(META1, META2)),
+                        cryptoApproveAllowance()
+                                .payingWith(DEFAULT_PAYER)
+                                .addNftAllowance(OWNER, NFT_TOKEN, AUTO_CREATION_MODES, false, List.of(2L))
+                                .via(BASE_APPROVE_TXN)
+                                .signedBy(DEFAULT_PAYER, OWNER)
+                                .fee(ONE_HBAR))
+                .when(withOpContext((spec, opLog) -> {
+                    final var ecdsaKey = spec.registry().getKey(ECDSA_KEY);
+                    final var tmp = ecdsaKey.getECDSASecp256K1().toByteArray();
+                    final var addressBytes = recoverAddressFromPubKey(tmp);
+                    allRunFor(
+                            spec,
+                            // transfer allowed NFT
+                            contractCall(
+                                            AUTO_CREATION_MODES,
+                                            "createIndirectlyRevertingAndRecover",
+                                            HapiParserUtil.asHeadlongAddress(
+                                                    asAddress(spec.registry().getTokenID(NFT_TOKEN))),
+                                            HapiParserUtil.asHeadlongAddress(
+                                                    asAddress(spec.registry().getAccountID(OWNER))),
+                                            HapiParserUtil.asHeadlongAddress(addressBytes),
+                                            2L)
+                                    .gas(GAS_TO_OFFER)
+                                    .via(TRANSFER_TXN)
+                                    .alsoSigningWithFullPrefix(OWNER)
+                                    .hasKnownStatus(SUCCESS));
+                }))
+                .then(
+                        getTxnRecord(TRANSFER_TXN).hasChildRecordCount(1),
```

Comment:
⚠️ We should use `.hasNonStakingChildRecordCount()` here to void flakiness in CI caused by the `TRANSFER_TXN` exporting an end-of-day node stake update transaction.

Actually it doesn't make sense to have two versions any more 😝 ...we should just delete `hasChildRecordCount()` and rename `hasNonStakingChildRecordCount()` to it.

Reply:
@tinker-michaelj I tried deleting `hasChildRecordCount()` and renaming `hasNonStakingChildRecordCount()` to it and this caused some specs to fail (please see below which). So I just used `.hasNonStakingChildRecordCount()` on the spec that I've added and propose to create another issue for `hasChildRecordCount()`.

Integration & HAPI Tests failures in `StakingSuite` for:
- `stakingMetadataUpdateIsRewardOpportunity`
- `SecondOrderRewardSituationsWork`
- `EndOfStakingPeriodRecTest`
- `RewardsOfDeletedAreRedirectedToBeneficiary`

Additional HAPI Test failure for `CongestionPricingSuite.java#canUpdateMultipliersDynamically2()`

[^ CI run](https://github.com/hashgraph/hedera-services/runs/21263076968)

Reply:
Ah that's a great point @thenswan , `StakingSuite` specifically wants to count both. 💯